tests/python/pants_test/backend/python/interpreter_selection_utils.py

```diff
@@ -107,3 +108,41 @@ def skip_unless_python27_and_python3_present(func):
 def skip_unless_python27_and_python36_present(func):
   """A test skip decorator that only runs a test method if python2.7 and python3.6 are present."""
   return skip_unless_all_pythons_present(PY_27, PY_36)(func)
+
+
+def skip_if_interpreter_is_any_python_version(*versions):
+  """A decorator that skips if the current interpreter version is any of the of the specified versions.
+
+  :param string *versions: Python version strings, such as "2.7", "3".
+  """
+  interpreter_major, interpreter_minor = sys.version_info[0:2]
+  parsed_versions = [version.split(".") for version in versions]
```

Comment:
I believe this could be simplified using `packaging.version.Version`: https://packaging.pypa.io/en/latest/version/. I think you could then do something like `packaging.version.Version(version).release == sys.version_info[0:2]`, removing the version matching helper method here.

Reply:
I couldn't get `packaging.version` working.

The main challenge with this function is that we must not only match major-minor like `3.6` and `2.7`, but we must also match only the major version like `3` and `2`. The suggested snippet `packaging.version.Version(version).release == sys.version_info[0:2]` wouldn't support this.

Even with keeping the current logic to conditionally check the minor version if present, for some reason I am getting the complaint `*** AttributeError: 'Version' object has no attribute 'release'`.

To reproduce, apply below diff and run `./pants test tests/python/pants_test/util:strutil -- -k test_camelcase`:

```diff
diff --git a/tests/python/pants_test/backend/python/BUILD b/tests/python/pants_test/backend/python/BUILD
index eee01488e..6cc8088b4 100644
--- a/tests/python/pants_test/backend/python/BUILD
+++ b/tests/python/pants_test/backend/python/BUILD
@@ -74,5 +74,6 @@ python_library(
   dependencies=[
     'src/python/pants/util:process_handler',
     '3rdparty/python:future',
+    '3rdparty/python:packaging',
   ]
 )
diff --git a/tests/python/pants_test/backend/python/interpreter_selection_utils.py b/tests/python/pants_test/backend/python/interpreter_selection_utils.py
index 46dbf63d4..fade36f5c 100644
--- a/tests/python/pants_test/backend/python/interpreter_selection_utils.py
+++ b/tests/python/pants_test/backend/python/interpreter_selection_utils.py
@@ -6,6 +6,7 @@ from __future__ import absolute_import, division, print_function, unicode_litera
 
 import os
 import sys
+from packaging.version import Version
 from unittest import skipIf
 
 from future.utils import PY2
@@ -116,14 +117,14 @@ def skip_if_interpreter_is_any_python_version(*versions):
   :param string *versions: Python version strings, such as "2.7", "3".
   """
   interpreter_major, interpreter_minor = sys.version_info[0:2]
-  parsed_versions = [version.split(".") for version in versions]
+  parsed_versions = [Version(version).release for version in versions]
 
   def version_matches_current_interpreter(major, minor=None):
-    if int(major) == interpreter_major and minor is None:
+    if major == interpreter_major and minor is None:
       return True
-    return int(major) == interpreter_major and int(minor) == interpreter_minor
+    return major == interpreter_major and minor == interpreter_minor
 
-  if any(version_matches_current_interpreter(*parsed_version) for parsed_version in parsed_versions):
+  if any(version_matches_current_interpreter(*parsed_version.release) for parsed_version in parsed_versions):
     return skipIf(True, "Current interpreter is one of the specified Python versions.")
   return skipIf(False, "Current interpreter")
 
diff --git a/tests/python/pants_test/util/BUILD b/tests/python/pants_test/util/BUILD
index b932c009e..8da5b4f55 100644
--- a/tests/python/pants_test/util/BUILD
+++ b/tests/python/pants_test/util/BUILD
@@ -166,6 +166,7 @@ python_tests(
   dependencies = [
     '3rdparty/python:future',
     'src/python/pants/util:strutil',
+    'tests/python/pants_test/backend/python:interpreter_selection_utils',
   ]
 )
 
diff --git a/tests/python/pants_test/util/test_strutil.py b/tests/python/pants_test/util/test_strutil.py
index f5a105a6f..d8cc05f73 100644
--- a/tests/python/pants_test/util/test_strutil.py
+++ b/tests/python/pants_test/util/test_strutil.py
@@ -8,11 +8,12 @@ import unittest
 from builtins import bytes
 
 from pants.util.strutil import camelcase, ensure_binary, ensure_text, pluralize, strip_prefix
-
+from pants_test.backend.python.interpreter_selection_utils import skip_if_python3
 
 # TODO(Eric Ayers): Backfill tests for other methods in strutil.py
 class StrutilTest(unittest.TestCase):
 
+  @skip_if_python3
   def test_camelcase(self):
 
     self.assertEqual('Foo', camelcase('foo'))
```

Reply:
Sorry for the difficulty trying to use this! It looks like `.release` is a newer feature, which we may want to upgrade `packaging` to handle  --  we use `packaging==16.8`, the master version is [`19.1.dev0`](https://github.com/pypa/packaging/blob/e78164be7cc250197a9b910955c5860257c86675/packaging/__about__.py#L21).

The reason I mentioned `Version` was because one thing I previously used in #7011 but removed was [`packaging.specifiers.SpecifierSet`](https://packaging.pypa.io/en/latest/specifiers/). It seems like what we're trying to do here might be more concisely expressed as interpreter constraint strings. With specifier sets, you can do:
```
> pex --python=python3 packaging==16.8
Python 3.6.4 (default, Jan 23 2018, 10:58:43)
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from packaging.specifiers import SpecifierSet
>>> '2.7' in SpecifierSet('==2.7')
True
>>> '2.7' in SpecifierSet('>=2.7')
True
>>> '3.7' in SpecifierSet('>=2.7,<3')
False
```

Reply:
We may want to consider drastically reducing the number of "skip if" decorators we have to just `@skip_if_python3`, `@skip_if_python2`, and `@skip_if_python_interpreter`.

Reply:
Much better now! Thank you!